src/pages/api/revalidate.ts

```diff
@@ -0,0 +1,79 @@
+import { NextApiRequest, NextApiResponse } from 'next/types'
+import { StatusCodes } from 'http-status-codes'
+import { validateToken } from 'lib/api-validate-token'
+import { cachedGetProductData } from 'lib/get-product-data'
+
+/**
+ * Accepts a POST request with a product slug, triggers revalidation for all of a product's paths
+ * specified in its latest nav data.
+ */
+export default validateToken(
+	async function handler(request: NextApiRequest, response: NextApiResponse) {
+		if (request.method !== 'POST') {
+			response.status(StatusCodes.NOT_FOUND)
```

Comment:
⛏️I was expecting `METHOD_NOT_ALLOWED`, but I also can understand using 404 if you don't want to expose that there _is_ a route at this path.

Reply:
Ooh good call! Yeah I like 404 here for that reason. Per the spec, we'd want to pass along the proper method in the `Allow` header in our response with the 405, which in turn exposes it. Not that it would be a major negative to implement it this way, but I don't think it provides any benefit for our uses.